src/Compilers/Test/Core/Compilation/OperationTreeVerifier.cs

```diff
@@ -62,13 +63,186 @@ public static string GetOperationTree(Compilation compilation, IOperation operat
 
         public static void Verify(string expectedOperationTree, string actualOperationTree)
         {
-            char[] newLineChars = Environment.NewLine.ToCharArray();
-            string actual = actualOperationTree.Trim(newLineChars);
-            actual = actual.Replace(" \n", "\n").Replace(" \r", "\r");
-            expectedOperationTree = expectedOperationTree.Trim(newLineChars);
-            expectedOperationTree = expectedOperationTree.Replace("\r\n", "\n").Replace(" \n", "\n").Replace("\n", Environment.NewLine);
+            // Finds spaces before newlines and places them in the specified toRemove buffer in reverse index order
+            static void FindSpacesBeforeNewlineHelper(char[] buffer, int bufferSize, ref int[] toRemove, ref int toRemoveCount)
```

Comment:
I really wonder whether the complex dance is measurably better than doing manipulation in `StringBuilder`.

Reply:
@jaredpar does this PR measurably improve your scenario?

What would the allocations for `StringBuilder.Replace` on .NET Framework be like @huoyaoyuan (which @jaredpar was concerned about iirc)? I'm not familiar with how it performs, as I usually am just `.Append`ing on `StringBuilder`s when I use them.

It seems like you'd at least have ~2x total string length in total, to store StringBuilder buffer & result string, whereas the current one is 3x for actual, and 4x for expected - so it seems like it could be at most ~2x better in the best scenario. Whereas this should be amortised allocation free (but obviously has a bit more complexity in implementation as a result).

Reply:
> @jaredpar does this PR measurably improve your scenario?

Going to be profiling this probably Tuesday. Monday tends to be pretty meeting heavy for me. Hoping to have time to profile this out tomorrow when I'm back working on tests.